platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/WorkNodeCodec.kt

```diff
@@ -91,6 +102,10 @@ class WorkNodeCodec(
             }
             nodes.add(node)
         }
+        readCollection {
+            // So far only task nodes can be entry nodes.
```

Comment:
💭 I'd prefer a solution that didn't involve adding state to the nodes (i.e. without a `markAsEntryNode` method and new boolean field) both because of the increased memory requirement (keeping a boolean on every task when just a small subset of them will ever be entry nodes), but also because it feels less intrusive and fragile.

For example, this method could return the two collections (the list of nodes and the list of entry nodes) which could be kept together in `CachedBuildState` and carried all the way to [`ConfigurationCacheState.calculateRootTaskGraph`](https://github.com/gradle/gradle/blob/b08b7722c1e9262ddf36316f3a838aa8cb50d8eb/platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheState.kt#L216) where they could be passed down as separate arguments to `setScheduledNodes`.

Reply:
> I'd prefer a solution that didn't involve adding state to the nodes

I did a rewrite. With the previous solution I avoided the pitfall of unscheduled entry nodes being present in the plan (by the virtue of only processing scheduled nodes when writing), but it looks like everything is still consistent. A mismatch between requested tasks and entry tasks is a bit concerning, though the solution I have now restores as much relevant state of the ExecutionPlan as possible, and the end result is the same.

Reply:
> With the previous solution I avoided the pitfall of unscheduled entry nodes being present in the plan (by the virtue of only processing scheduled nodes when writing)

I see. I still prefer the new solution with the proper javadocs.